addons/account/models/account_move.py

```diff
@@ -391,6 +391,20 @@ def _sequence_fixed_regex(self):
         required=True,
         compute='_compute_currency_id', inverse='_inverse_currency_id', store=True, readonly=False, precompute=True,
     )
+    exchange_rate = fields.Float(
+        string="Currency rate",
+        tracking=True,
+        copy=False,
```

Comment:
We don't want any extra field stored btw ;)

Reply:
@smetl what do you mean? what happens if you reset to draft and change the qty?

Reply:
The rate is just there to be applied once. Otherwise, you compute it directly from lines like a normal compute / inverse field. Storing it is pointless.

Reply:
@smetl The whole idea is that you can set a manual rate.  And the whole balance vs amount_currency does give rounding errors.

Reply:
Ideally, we would have a rate and a boolean which says 'manual override'...

Reply:
What happens if you set the rate, then add lines?
Then save, then add some more lines?